mycore-base/src/test/java/org/mycore/common/util/MCRTestCaseXSLTUtil.java

```diff
@@ -32,28 +36,50 @@
 import javax.xml.transform.stream.StreamSource;
 import java.util.Map;
 
+/**
+ * Provides utility methods for testing XSLT functions. The general process for XSL testing is as follows:
+ * <ol>
+ *     <li>create a test XSL file containing match templates calling functions/templates to test</li>
+ *     <li>call prepareTestDocument with the rootName matching the test template (and possible XML content)</li>
+ *     <li>call transform with the test document and evaluate the result document</li>
+ * </ol>
+ */
 public class MCRTestCaseXSLTUtil {
+    private static final Logger LOGGER = LogManager.getLogger();
+    private static final String ERROR_XTMM9000 = "XTMM9000";
+    private static final String ERROR_XTMM9001 = "XTMM9001";
+
+    /**
+     * Returns an XML document with a root element with the given name
+     *
+     * @param rootName the root element name of the document
+     * @return an XML document with a root element with the given name
+     */
+    public static Document prepareTestDocument(String rootName) {
+        return new Document().addContent(new Element(rootName));
+    }
+
     /**
-     * Returns an XML document with the given method name as root element and the content of the given XML tree as
+     * Returns an XML document with the given name as root element and the content of the given XML tree as
      * child element. The resulting document can be used for calling XSLT test files using template matching.
      *
-     * @param method the method name used as root element name
-     * @param xml    the XML tree
-     * @return an XML document with the method name as root element and the content of the given XML tree as children
+     * @param rootName the name used as root element name
+     * @param xml      the XML tree
+     * @return an XML document with the name as root element and the content of the given XML tree as children
      */
-    public static Document xmlTest(String method, Element xml) {
-        final Element root = new Element(method);
+    public static Document prepareTestDocument(String rootName, Element xml) {
```

Comment:
Idea: to be more flexible: the 2nd param could be a Content list (TextNodes or Elements)
or provide another prepareTestDocument() method ?